proto/storage/compliance_operator_v2.proto

```diff
@@ -46,33 +69,75 @@ message ComplianceOperatorRuleV2 {
+    map<string, string>           labels = 5;
+    map<string, string>           annotations = 6;
+    repeated Profile              profiles = 7;
+    repeated ClusterScanStatus    clusters = 8;
+    repeated NodeRole             node_roles = 9;
+    // Will be configurable via env var
+    bool                          strict_node_scan = 10;
     // Starting point for schedule will probably have to build upon it
-    storage.Schedule              schedule                 = 11;
-    google.protobuf.Timestamp     created_time             = 12;
-    google.protobuf.Timestamp     last_updated_time        = 13;
+    storage.Schedule              schedule = 11;
+    google.protobuf.Timestamp     created_time = 12;
+    google.protobuf.Timestamp     last_updated_time = 13;
     // Most recent user to update the scan settings
-    SlimUser                      created_by               = 14;
+    SlimUser                      created_by = 14;
+}
+
+// Next Tag: 15
+// This object has been flattened vs joining with rule.  The rationale is to spend the time to query rule
+// while processing results vs reporting them to the user.  Additionally, flattening it helps with the historical data
+// as the rules can change without impacting the historical result.
+message ComplianceOperatorCheckResultV2 {
+    enum CheckStatus {
+        UNSET = 0;
+        PASS = 1;
+        FAIL = 2;
+        ERROR = 3;
+        INFO = 4;
+        MANUAL = 5;
+        NOT_APPLICABLE = 6;
+        INCONSISTENT = 7;
+    }
+    string              id = 1 [(gogoproto.moretags) = 'sql:"pk"'];
+    string              check_id = 2;
+    string              check_name = 3;
+    string              cluster_id = 4 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid)"'];
+    CheckStatus         status = 5  [(gogoproto.moretags) = 'search:"Compliance Check Status,hidden,store"'];
+    string              severity     = 6 [(gogoproto.moretags) = 'search:"Compliance Operator Severity,hidden,store"'];
+    string              description = 7;
+    string              instructions = 8;
+    map<string, string> labels = 9;
+    map<string, string> annotations = 10;
+    google.protobuf.Timestamp     created_time = 11;
+    string standard = 12 [(gogoproto.moretags) = 'search:"Standard,hidden,store"'];
+    string control = 13;
+    string scan_id = 14 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorScanV2:id),no-fk-constraint"'];
+}
+
+message ScanStatus {
+    string phase = 1;
+    string result = 2;
+    repeated string warnings = 3;
+}
+
+// Next Tag: 13
+// Scan object per cluster
+message ComplianceOperatorScanV2 {
+    string              id          = 1 [(gogoproto.moretags) = 'sql:"pk"'];
+    string              scan_name        = 2 [(gogoproto.moretags) = 'search:"Compliance Operator Scan Name,hidden,store" sql:"index=category:unique;name:scan_unique_indicator"'];
```

Comment:
fk? (this is bound to scan setting binding)

Reply:
good catch